### ISSUE TITLE
client support for SFTP

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7236]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| z3p reported|
|-|-|
|Trac ID|trac#497|
|Type|enhancement|
|Created|2004-02-11 10:25:52Z|

<details><summary>Searchable metadata</summary>

```
trac-id__497 497
type__enhancement enhancement
reporter__z3p z3p
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1076495152000000 1076495152000000
changetime__1471873386080428 1471873386080428
version__ 
owner__ 
cc__z3p cc__spiv cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
You're doing this now aren't you? Assigned.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|z3p commented|
|-|-|

```
#!html
<pre>
r11939 is the first commit of cftp.
outstanding issues:
* better text from get/put
* multiple outstanding get/put requests
* !command and ! support
* probably other things that other people will catch
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="z3p's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|z3p commented|
|-|-|

```
#!html
<pre>
as of r11973, batchfile and better text from get/put are fixed.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

So.... is this resolved?  The ultimate goal of any ticket is to be resolved.  It looks like some work was done torwards "client support for SFTP".  Is this ticket resolved?

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

Without actually knowing much about the underlying SFTP client code, I suggest closing this ticket.  We can file specific tickets for and specific issues that get reported, but a ticket this vague doesn't really achieve much; it doesn't indicate where we are, or where a willing volunteer should start work to improve things.

Maybe if there were a big suite of "client support for SFTP acceptance tests" or other useful criteria in the ticket to know when it was done, my opinion would be different.

Anyway, my impression is that the underlying infrastructure for client SFTP basically exists, and there's a proof-of-concept client that basically works.  So we should close this.  If there are missing features or bugs, those can be new tickets.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_8"></a>|@rodrigc commented|
|-|-|

In changeset b3bf0d24f5c0bd6e7ea44369abff1d395ef18c53
```
#!CommitTicketReference repository="" revision="b3bf0d24f5c0bd6e7ea44369abff1d395ef18c53"
Merge pull request #497 from twisted/8758-rodrigc-iutils-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8758

Reviewer: Fix test_iutils.py on Windows Python 3
```
